pkg/profilecreator/profilecreator.go

```diff
@@ -273,10 +303,17 @@ func (ghwHandler GHWHandler) GetReservedAndIsolatedCPUs(reservedCPUCount int, sp
 	if err != nil {
 		return cpuset.CPUSet{}, cpuset.CPUSet{}, fmt.Errorf("can't determine if Hyperthreading is enabled or not: %v", err)
 	}
+	//currently HT is enabled on the system and the user wants to disable HT
+	if htEnabled && disableHTFlag {
+		log.Infof("Currently hyperthreading is enabled and the performance profile will disable it")
+		topologyInfo = ghwHandler.TopologyHTDisabled(topologyInfo)
+
```

Comment:
so here inside this `if` we can now do
```
htEnabled =  false
```
and just use `htEnabled` from now on, without the need of carrying with us two separate flags, right?

Reply:
We need two separate flags as `htEnabled` represents the current state of hyperthreading on the system and `disableHTFlag` represents the intended state of the HT that the user wants the system to be in.

Reply:
ok, but where this difference materializes? in both `getCPUsSplitAcrossNUMA` and `getCPUsSequentially` I always see the flag being used like `htEnabled && !disableHTFlag`

Reply:
I understand now what you mean!
The reason why it is implemented like this is as follows:
1. In case hyperthreading is enabled on the system (htEnabled = true) and the user does not want to disable it (disableHTFlag=false) we can't allow odd number of CPUs to be allocated from a NUMA node.
In rest of the cases (as follows) we can have odd number of CPUs to be allocated from a NUMA node
2. htEnabled = false disableHTFlag= false
3. htEnabled = true disableHTFlag= true
4. htEnabled = false disableHTFlag= true

but what you are saying makes sense now as in case 3 if assign htEnable=false and that would mean that  htEnabled && !disableHTFlag would return false for all case 2, 3 and 4 (achieving the behavior we want).